src/Bicep.Core/Parser/Parser.cs

```diff
@@ -297,25 +303,112 @@ private SyntaxBase ParenthesizedExpression()
 
         private SyntaxBase FunctionCallOrVariableAccess()
         {
-            var identifier = this.Identifier(b => b.ExpectedFunctionName());
+            if (IsFunctionCall())
```

Comment:
>IsFunctionCall [](start = 16, length = 14)

If the token at current position is not an identifier, this will try to parse it as VariableAccess. Can it result in misleading error messages?